packages/api/amplify_api/lib/src/method_channel_api.dart

```diff
@@ -1,379 +0,0 @@
-
-import 'package:amplify_api/src/graphql/graphql_response_decoder.dart';
-import 'package:amplify_api/src/graphql/graphql_subscription_event.dart';
-import 'package:amplify_api/src/graphql/graphql_subscription_transformer.dart';
-import 'package:amplify_core/amplify_core.dart';
-
-import 'package:flutter/foundation.dart';
-import 'package:flutter/services.dart';
-
-import '../amplify_api.dart';
-
-part 'auth_token.dart';
-
-const MethodChannel _channel = MethodChannel('com.amazonaws.amplify/api');
-
-/// {@template amplify_api.amplify_api_method_channel}
-/// The method channel implementation of [AmplifyAPI].
-///
-/// Used on iOS and Android clients.
-/// {@endtemplate}
-class AmplifyAPIMethodChannel extends AmplifyAPI {
-  /// {@macro amplify_api.amplify_api_method_channel}
-  AmplifyAPIMethodChannel({
-    List<APIAuthProvider> authProviders = const [],
-    this.modelProvider,
-  }) : super.protected() {
-    authProviders.forEach(registerAuthProvider);
-  }
-
-  static const _eventChannel =
-      EventChannel('com.amazonaws.amplify/api_observe_events');
-  late final Stream<GraphQLSubscriptionEvent> _allSubscriptionsStream =
-      _eventChannel
-          .receiveBroadcastStream(0)
-          .transform(_graphQLSubscriptionEventTransformer);
-
-  /// Transforms incoming platform events into typed [GraphQLSubscriptionEvent] objects
-  /// and relays errors to the correct subscription stream.
-  late final _graphQLSubscriptionEventTransformer =
-      StreamTransformer<dynamic, GraphQLSubscriptionEvent>.fromHandlers(
-    handleData: (
-      dynamic data,
-      EventSink<GraphQLSubscriptionEvent> eventSink,
-    ) {
-      final eventJson = (data as Map).cast<String, dynamic>();
-      eventSink.add(GraphQLSubscriptionEvent.fromJson(eventJson));
-    },
-    handleError: (
-      Object error,
-      StackTrace stackTrace,
-      EventSink<GraphQLSubscriptionEvent> eventSink,
-    ) {
-      if (error is! PlatformException) {
-        eventSink.addError(error, stackTrace);
-        _subscriptions.clear();
-        return;
-      }
-      final subscriptionId = error.message;
-      if (subscriptionId == null ||
-          !_subscriptions.containsKey(subscriptionId)) {
-        eventSink.addError(error, stackTrace);
-        _subscriptions.clear();
-        return;
-      }
-      final errorEvent = GraphQLSubscriptionEvent(
-        subscriptionId: subscriptionId,
-        type: GraphQLSubscriptionEventType.error,
-        error: _deserializeException(error),
-      );
-      eventSink.add(errorEvent);
-    },
-  );
-
-  final Map<String, Stream<GraphQLResponse>> _subscriptions = {};
-
-  /// The registered [APIAuthProvider] instances.
-  final Map<APIAuthorizationType, APIAuthProvider> _authProviders = {};
-
-  @override
-  final ModelProviderInterface? modelProvider;
-
-  @override
-  Future<void> addPlugin() async {
-    try {
-      setupAuthProviders();
-      await _channel.invokeMethod<void>('addPlugin', {
```

Comment:
callout: datastore integration tests fail, I think bc needs this. Will eventually need some way to call this when using with datastore on supported platforms until datastore implemented.